lib/ecmascript.ts

```diff
@@ -1243,86 +1296,24 @@ export function PrepareTemporalFields<
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         value = BUILTIN_CASTS.get(property)!(value);
       }
-      result[property] = value as string | number | undefined;
-    } else if (completeness !== 'partial') {
-      if (wrappedFieldRecord.length === 1) {
+      result[property] = value;
+    } else if (requiredFields !== 'partial') {
+      // TODO: using .call in this way is not correctly type-checked by tsc.
+      // We might need a type-safe Call wrapper?
```

Comment:
Could es-abstract's Call be the thing to use here?

Reply:
The typings for es-abstract's Call look useful. I'll investigate this and follow up with a PR once I've figured out if there's enough of these mis-typed calls in our codebase to justify using the es-abstract version or whether a local function definition is enough.